nsxt/resource_nsxt_manager_cluster.go

```diff
@@ -314,6 +258,34 @@ func getHostCredential(m interface{}) (string, string) {
 	return username, password
 }
 
+func getMatchingIPVersion(ip string, hostIPs []string) string {
+	needIPv4 := (net.ParseIP(ip)).To4() != nil
+
+	for _, hostIP := range hostIPs {
+		isIPv4 := (net.ParseIP(hostIP)).To4() != nil
+		if needIPv4 == isIPv4 {
```

Comment:
The code looks good but I find some difficulties to read it. It's not easy to understand that if needIPv4=False and isIPv4=False then we want to return an IPv6 address?

Do you agree or is it just me getting older?

Reply:
Its a bit tricky to read, I agree, I've added a comment